pageserver/src/tenant.rs

```diff
@@ -2338,17 +2341,18 @@ impl Tenant {
             src_timeline.initdb_lsn,
             src_timeline.pg_version,
         );
-        let mut timelines = self.timelines.lock().unwrap();
-        let new_timeline = self
-            .prepare_timeline(
+
+        let new_timeline = {
+            let mut timelines = self.timelines.lock().unwrap();
+            self.prepare_timeline(
                 dst_id,
                 &metadata,
                 timeline_uninit_mark,
                 false,
                 Some(Arc::clone(src_timeline)),
             )?
-            .initialize_with_lock(ctx, &mut timelines, true, true)?;
-        drop(timelines);
+            .initialize_with_lock(ctx, &mut timelines, true, true)?
+        };
```

Comment:
Massage required to avoid NLL thinking the `timelines` is alive after `drop(timelines)`.

Reply:
Not required anymore after [2d2c5af](https://github.com/neondatabase/neon/pull/4204/commits/2d2c5af072ef1a13a35839a0b910f09feb4b7511) but would prefer to keep it.